docs: Review Developer docs (#4776)
recognai/rubrix: docs/_source/community/developer_docs.md

- **Vue.js UI**: A web application to visualize and annotate your data, users, and teams. It is built with `Vue.js` and is directly deployed alongside the Argilla Server within our Argilla Docker image.

> The Argilla repository has a mono repo structure, which means that all the components are in the same repository, and is divided into the following folders:

Reviewer: ```suggestion
The Argilla repository has a monorepo structure, which means that all the components live in the same repository: `argilla-io/argilla`. This repo is divided into the following folders:
```

Then, you just need to install Argilla with the command below. Note that we will install it in editable mode using the -e/--editable flag in the `pip` command to avoid having to re-install it on every code modification, but if you’re not planning to modify the code, you can just omit the -e/--editable flag.

```sh

> cd argilla

Reviewer: Do we really need to add `cd argilla` in this and upcoming blocks? The first block after "Install with `pip`" already covers this step.

After deleting the database, you will need to run the [database migration](#run-database-migration) task. By following these steps, you’ll have a fresh and clean database to work with.

### Set up Argilla Server

> If you want to work on the server of Argilla, please visit the `argilla-server` [README.md](./argilla-server/README.md) file to see how to set up the server and run it on your local machine.

Reviewer: Seems like the link isn't working.

### Set up Argilla Frontend

> If you want to work on the frontend of Argilla, please visit the `argilla-frontend` [README.md](./argilla-frontend/README.md) file to see how to set up the frontend and run it on your local machine.

Reviewer: This link isn't working either.

## Make Your Contribution

> Now that everything is up and running, you can start to develop and contribute to Argilla! You can refer to our [contributer guide](/community/contributing.md) to have an understanding of how you can structure your contribution and upload it to the repository.

Reviewer: ```suggestion
Now that everything is up and running, you can start to develop and contribute to Argilla! You can refer to our [contributer guide](/community/contributing.md) to have an understanding of how you can structure your contribution and upload it to the repository.
```
```suggestion
Now that everything is up and running, you can start to develop and contribute to Argilla! You can refer to our [contributor guide](/community/contributing.md) to have an understanding of how you can structure your contribution and upload it to the repository.
```
